wavedump/src/fit-histograms

```diff
@@ -0,0 +1,566 @@
+    n -= h.GetBinContent(i)
+    i -= 1
+    low_prob = n / total
+    l = (((0.25 - low_prob) / (high_prob - low_prob)) * h.GetBinWidth(i)) + h.GetBinLowEdge(i)
+    while n / total < 0.75:
+        i += 1
+        n += h.GetBinContent(i)
+    high_prob = n / total
+    n -= h.GetBinContent(i)
+    i -= 1
+    low_prob = n / total
+    r = (((0.75 - low_prob) / (high_prob - low_prob)) * h.GetBinWidth(i)) + h.GetBinLowEdge(i)
+    return r-l
+
+def average_distance_from(val, h):
+    """
+    Standard deviation calculation, except use `val` instead of the mean.
+    """
+    d2 = 0
+    for i in range(h.GetMaximumBin() + 2):
+        d2 += (val - h.GetXaxis().GetBinCenter(i))**2 * h.GetBinContent(i)
+    return np.sqrt(d2)
+
+def save_as_csv(data_type, value, value_err, channel):
+    """
+    Saves charge data in `temp_data.csv`.
+    `temp_data.csv` is read by `save_spe_data.py`, which
+    organizes the data and saves it to `Fit_Data.csv`.
+    """
+    with open ('temp_data.csv', 'a') as file:
+        headers = [
+            'Data Type',
+            'Channel',
+            'Charge',
+            'Err Abs'
+            ]
+        writer = csv.DictWriter(file, headers)
+        if os.stat('temp_data.csv').st_size == 0:
+            writer.writeheader()
+        writer.writerow({
+            'Data Type': data_type,
+            'Channel': channel,
+            'Charge': value,
+            'Err Abs': value_err
+        })
+
+def remove_bad_peaks(h, x_pos, win):
+    """
+    At each value (`a`) in `x_pos`, fits a gaussian to `h` in the window `(a - win, a + win)`.
+    If the mean of the fitted gaussian is less than or equal to `win * 0.25` away from `a`, then `a` is called a good peak.
+    
+    Returns an array of all the good peaks in `x_pos`.
+    """
+    rm = []
+    for i in range(len(x_pos)):
+        ge = ROOT.TF1("ge", 'gaus', x_pos[i] - win, x_pos[i] + win)
+        ge.SetLineColor(3)
+        if i == 0:
+            r = h.Fit(ge, 'SRQ')
+        else:
+            r = h.Fit(ge, 'SRQ+')
+        print(f'Peak {i} difference: {np.abs(x_pos[i] - ge.GetParameter(1))}')
+        if np.abs(x_pos[i] - ge.GetParameter(1)) > win * 0.25:
+            rm.append(i) 
+    x_pos = np.delete(x_pos, rm)
+    return x_pos
+
+def analyze_filter_data(h, f_h):
+    """
+    Fits the filtered charge histogram `f_h` with a gaussian,
+    and gets the mean and standard  deviation of this gaussian: `f_m` and `f_std`.
+    
+    Then, fits `h` (the unfiltered charge histogram) with a gaussian around `f_m`,
+    and gets the mean, standard deviation, and scale of this gaussian: `m`, `std`, `scale`.
+
+    Returns `(m, f_std, std, scale)`
+
+    If `f_h` is passed as `None`, only default values are returned, specified as global constants
+    in this module (except for the default scale, which is proportional to the entries in `h`).
+    """
+    if f_h == None:
+        return (D_OFFSET, D_NOISE_SPREAD, D_ZERO_PEAK_SPREAD, h.GetEntries()*0.075)
```

Comment:
Not super important, but I think this behavior is a little confusing. I would rather that in the code below if you don't have access to the filtered data you just use the default parameters there and then have this function only called if you use the filtered data.